File: platforms/android/Dezel/dezel/src/main/java/ca/logaritm/dezel/modules/platform/JavaScriptPlatform.kt
package ca.logaritm.dezel.modules.platform

import android.os.Build
import ca.logaritm.dezel.core.JavaScriptClass
import ca.logaritm.dezel.core.JavaScriptContext
import ca.logaritm.dezel.core.JavaScriptGetterCallback

/**
 * @class JavaScriptPlatform
 * @since 0.7.0
 * @hidden
 */
open class JavaScriptPlatform(context: JavaScriptContext): JavaScriptClass(context) {

	//--------------------------------------------------------------------------
	// JS Functions
	//--------------------------------------------------------------------------

	/**
	 * @method jsGet_name
	 * @since 0.7.0
	 * @hidden
	 */
	open fun jsGet_name(callback: JavaScriptGetterCallback) {
		callback.returns("android")
	}

	/**
	 * @method jsGet_version
	 * @since 0.7.0
	 * @hidden
	 */
	open fun jsGet_version(callback: JavaScriptGetterCallback) {
		callback.returns(Build.VERSION.RELEASE)
	}
}

File: platforms/ios/Dezel/Dezel/Modules/Platform/JavaScriptPlatform.swift
/**
 * @class JavaScriptPlatform
 * @super JavaScriptClass
 * @since 0.7.0
 */
open class JavaScriptPlatform: JavaScriptClass {

	//--------------------------------------------------------------------------
	// MARK: JS Properties
	//--------------------------------------------------------------------------

	/**
	 * @method jsGet_name
	 * @since 0.7.0
	 * @hidden
	 */
	@objc func jsGet_name(callback: JavaScriptGetterCallback) {
		callback.returns("ios")
	}

	/**
	 * @method jsGet_version
	 * @since 0.7.0
	 * @hidden
	 */
	@objc func jsGet_version(callback: JavaScriptGetterCallback) {
		callback.returns(UIDevice.current.systemVersion)
	}
}
